vivian_airflow_extensions/operators/snowflake_to_postgres_operator.py

```diff
@@ -102,29 +127,38 @@ def execute(self, context):
                 self.columns_list = self.postgres_hook.get_table_metadata(self.postgres_table, self.schema, self.include_autoincrement_keys)
                 self.metadata_retrieved = True
                 self.columns_to_update = [col for col in self.columns_list if col not in self.primary_key_columns]
-                
-            columns_to_update_string = ", ".join([f'"{col}"=excluded."{col}"' for col in self.columns_to_update])
+                self._validate_insert_only_columns()
```

Comment:
why are you doing this twice?

Reply:
Because if columns_to_update isn't set as a param, it gets set on L129/L104, right above here

Reply:
But if columns to update is none this will fail anyway during init, no?

Reply:
No, see L111 https://github.com/nursefly/nursefly-analytics/blob/50ac5394d3a90eaeb9162368e3e33e9741ada21a/nursefly-dwh/models/marts/jobs/intermediate/_job_history.sql#L5, the check gets skipped if columns_to_update isn't set yet.

Reply:
The point is, this should work if you don't specify columns_to_update, but it'll still check at runtime that your insert_only_columns actually exist in the table.

Reply:
why not just check it once by pulling it out of the if statement that starts on L126?

Reply:
I thought it would be preferable to fail at DAG compile time (ie in the constructor) if possible, and only fail at execution time if necessary. But I will simplify it.